Packages/Core/Editor/Generator/Generator.cs

```diff
@@ -21,6 +22,191 @@ public static string[] GetTemplatePaths()
             return Directory.GetFiles(templateSearchPath, "UA_Template*.txt", SearchOption.AllDirectories);
         }
 
+        public static Dictionary<string, string> CreateTemplateVariablesMap(Type type, bool asPair, string withNamespace = default)
+        {
+            var typeName = type.FullName.Replace('+', '.'); // This accounts for nested types.
+            var displayNameNoPair = type.Name.Capitalize();
+            var displayName = displayNameNoPair;
+
+            if (asPair)
+            {
+                typeName = $"Pair<{typeName}>";
+                displayName += "Pair";
+            }
+
+            // Add Types.
+            var templateVariables = new Dictionary<string, string>()
+            {
+                { "VALUE_TYPE", typeName },
```

Comment:
Hard-coded string to enums, would it make sense?

Reply:
These methods actually allow you to use the old generator with types, which was my first approach. These will also be deprecated (which is funny cause they’re new) before I make this pr ready for review.